applications: nrf5340_audio: LED module for kits with other LED setup (#20483)
nrfconnect/sdk-nrf: applications/nrf5340_audio/doc/user_interface.rst

| Switch            | Function                                                                            | Applications                          | Development Kit               |
|                   |                                                                                     |                                       +------------------+------------+
|                   |                                                                                     |                                       | nRF5340 Audio DK | nRF5340 DK |
+===================+=====================================================================================+=======================================+==================+============+
| **POWER**         | Turns the development kit on or off.                                                | All                                   |         Y        |      Y     |
+-------------------+-------------------------------------------------------------------------------------+---------------------------------------+------------------+------------+

> | **DEBUG ENABLE**  | Turns on or off power for debug features.                                           | All                                   |         Y        |      N     |

Reviewer: https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/applications/nrf5340_audio/doc/requirements.html are not mentioning the nRF5340 DK as supported platform for the audio applications, so you cannot add mentions of the nRF5340 DK to the docs until that page is updated. This is because the documentation will become inconsistent and confusing to users.

Author: Added nRF5340 DK as a supported platform.

Reviewer: Unfortunately, we need to revert to the old formatting. See my other comment.

The application uses the following buttons on the supported development kit:

+---------------+-----------------------------------------------------------------------------------------------------------+---------------------------------------------+-------------------------------+
| Button        | Function                                                                                                  | Applications                                | Development Kit               |
|               |                                                                                                           |                                             +------------------+------------+

> |               |                                                                                                           |                                             | nRF5340 Audio DK | nRF5340 DK |

Reviewer: This addition is not matching the list of supported boards in Requirements.

Author: As above.

+===============+===========================================================================================================+=============================================+==================+============+

> | **VOL-**      | Long-pressed during startup: Changes the headset to the left channel one.                                 | * :ref:`nrf53_audio_broadcast_sink_app`     |         Y        |      Y     |

Reviewer 2: How do you want this to look @greg-fer: Now the buttons are in the leftmost column. We could delete that column all together and put the button names in the two rightmost columns.

Reviewer: If we have time, I'd suggest using tabs for different kits like here https://github.com/nrfconnect/sdk-nrf/blame/main/samples/bluetooth/peripheral_uart/README.rst#L112 . We get less crowded tables and clear reference for each kit.

Author: I like that!

|               | The available values are 1000 Hz, 2000 Hz, and 4000 Hz.                                                   |                                             |                  |            |
|               +-----------------------------------------------------------------------------------------------------------+---------------------------------------------+------------------+------------+
|               | Pressed on a BIS headset during playback: Change stream (different BIS), if more than one is available.   | :ref:`nrf53_audio_broadcast_sink_app`       |         Y        |      N     |
+---------------+-----------------------------------------------------------------------------------------------------------+---------------------------------------------+------------------+------------+
| **BTN 5**     | Long-pressed during startup: Clears the previously stored bonding information.                            | * :ref:`nrf53_audio_unicast_server_app`     |         Y        |      Y     |

> | **Buton 4**   |                                                                                                           | * :ref:`nrf53_audio_unicast_client_app`     |                  |            |

Reviewer 2: ```suggestion
| **Button 4**   |                                                                                                           | * :ref:`nrf53_audio_unicast_client_app`     |                  |            |
```

To indicate the tasks performed, the application uses the LED behavior described in the following table:

+--------------------------+-----------------------------------------------------------------------------------------------------------+---------------------------------------------+-------------------------------+
| LED                      |Indication                                                                                                 | Applications                                | Development Kit               |
|                          |                                                                                                           |                                             +------------------+------------+

> |                          |                                                                                                           |                                             | nRF5340 Audio DK | nRF5340 DK |

Reviewer: Same comment as above.

Author: As above.

|                          | Solid green - Sync achieved (both drift and presentation compensation are in the ``LOCKED`` state).       | * :ref:`nrf53_audio_broadcast_sink_app`     |         Y        |      Y     |
|                          |                                                                                                           | * :ref:`nrf53_audio_unicast_server_app`     |                  |            |
+--------------------------+-----------------------------------------------------------------------------------------------------------+---------------------------------------------+------------------+------------+
| **LED3**                 | Blinking green - The nRF5340 application core is running.                                                 | All                                         |         Y        |      Y     |
+--------------------------+-----------------------------------------------------------------------------------------------------------+---------------------------------------------+------------------+------------+

> | **LED5**                 | Off - No PC connection available.                                                                         | All                                         |         N        |      Y     |

Reviewer 2: This LED also exists on the 5340 DK. Discuss as above if we should write "IFMCU" in the Audio DK column and "LED5" in the vanilla DK column.

Also, this info is not that crucial. @greg-fer, do we usually document this IFMCU/Segger behavior on each DK?

Reviewer: There are no guidelines about not documenting it. User Interface should document the behavior of the LEDs and buttons for the purpose of running and testing the application. If it needs mentioning SEGGER components, we can do it.

|                          +-----------------------------------------------------------------------------------------------------------+                                             +------------------+------------+
|                          | Solid green - Connected to PC.                                                                            |                                             |         N        |      Y     |
|                          +-----------------------------------------------------------------------------------------------------------+                                             +------------------+------------+

> |                          | Rapid flashing green - Device being flashed.                                                              |                                             |         N        |      Y     |

Reviewer 2: ```suggestion
|                          | Rapid flashing green - IFMCU/Debugger operation ongoing.                                                               |                                             |         N        |      Y     |
```

+--------------------------+-----------------------------------------------------------------------------------------------------------+---------------------------------------------+------------------+------------+

> | **LED1** to **LED4**     | All solid green - Fault in the application core has occurred.                                             | All                                         |         N        |      Y     |

Reviewer 2: For clarity, I suggest: 
```suggestion
| **LED1,2,3 and 4**    | All solid green - Fault in the application core has occurred.                                             | All                                         |         N        |      Y     |
```

|                          | Solid green - On-board hardware codec selected.                                                           |                                             |                  |            |
|                          +-----------------------------------------------------------------------------------------------------------+                                             +------------------+------------+
|                          | Solid yellow - External hardware codec selected.                                                          |                                             |                  |            |
|                          | This LED turns solid yellow also when the devices are reset, for the time then pins are floating.         |                                             |                  |            |
+--------------------------+-----------------------------------------------------------------------------------------------------------+---------------------------------------------+------------------+------------+

> | **FTDI SPI**             | Off - No data is written to the hardware codec using SPI.                                                 | All                                         |         Y        |      N     |

Reviewer 2: General: To ease readability: Do we need subrows for "y"/"n" within the same LED?

Reviewer: No, this can also be solved with tabs, as here: https://github.com/nrfconnect/sdk-nrf/blame/main/samples/bluetooth/peripheral_uart/README.rst#L112
